evennia/utils/tests/test_text2html.py

```diff
@@ -0,0 +1,187 @@
+import mock
+
+
+class TestText2Html(TestCase):
+
+    def test_re_color(self):
+        parser = text2html.HTML_PARSER
+        self.assertEqual("foo", parser.re_color("foo"))
+        self.assertEqual(
+            "<span class=\"color-001\">red</span>foo",
+            parser.re_color(ansi.ANSI_UNHILITE + ansi.ANSI_RED + "red" + ansi.ANSI_NORMAL + "foo"))
+        self.assertEqual(
+            "<span class=\"bgcolor-001\">red</span>foo",
+            parser.re_color(ansi.ANSI_BACK_RED + "red" + ansi.ANSI_NORMAL + "foo"))
+        self.assertEqual(
+            "<span class=\"bgcolor-001\"><span class=\"color-002\">red</span></span>foo",
+            parser.re_color(ansi.ANSI_BACK_RED + ansi.ANSI_UNHILITE + ansi.ANSI_GREEN + "red" + ansi.ANSI_NORMAL + "foo"))
+
+    @unittest.skip("parser issues")
+    def test_re_bold(self):
+        parser = text2html.HTML_PARSER
+        self.assertEqual("foo", parser.re_bold("foo"))
+        self.assertEqual(
+            # "a <strong>red</strong>foo",  # TODO: why not?
+            "a <strong>redfoo</strong>",
+            parser.re_bold(
+                "a " + ansi.ANSI_HILITE + "red" + ansi.ANSI_UNHILITE + "foo"))
+
+    @unittest.skip("parser issues")
+    def test_re_underline(self):
+        parser = text2html.HTML_PARSER
+        self.assertEqual("foo", parser.re_underline("foo"))
+        self.assertEqual(
+            "a <span class=\"underline\">red</span>" + ansi.ANSI_NORMAL + "foo",
+            parser.re_underline(
+                "a " + ansi.ANSI_UNDERLINE + "red"
+                + ansi.ANSI_NORMAL  # TODO: why does it keep it?
+                + "foo"))
+
+    @unittest.skip("parser issues")
+    def test_re_blinking(self):
+        parser = text2html.HTML_PARSER
+        self.assertEqual("foo", parser.re_blinking("foo"))
+        self.assertEqual(
+            "a <span class=\"blink\">red</span>" + ansi.ANSI_NORMAL + "foo",
+            parser.re_blinking(
+                "a " + ansi.ANSI_BLINK + "red"
+                + ansi.ANSI_NORMAL  # TODO: why does it keep it?
+                + "foo"))
+
+    @unittest.skip("parser issues")
+    def test_re_inversing(self):
+        parser = text2html.HTML_PARSER
+        self.assertEqual("foo", parser.re_inversing("foo"))
+        self.assertEqual(
+            "a <span class=\"inverse\">red</span>" + ansi.ANSI_NORMAL + "foo",
+            parser.re_inversing(
+                "a " + ansi.ANSI_INVERSE + "red"
+                + ansi.ANSI_NORMAL  # TODO: why does it keep it?
+                + "foo"))
+
+    @unittest.skip("parser issues")
+    def test_remove_bells(self):
+        parser = text2html.HTML_PARSER
+        self.assertEqual("foo", parser.remove_bells("foo"))
+        self.assertEqual(
+            "a red" + ansi.ANSI_NORMAL + "foo",
+            parser.remove_bells(
+                "a " + ansi.ANSI_BEEP + "red"
+                + ansi.ANSI_NORMAL  # TODO: why does it keep it?
+                + "foo"))
```

Comment:
This is correct behavior and not a bug. An `ANSI_BEEP` is just a single instruction and has no start/end tag. We should just clean it out and not touch the `ANSI_NORMAL` (which is likely terminating another, earlier instruction).